Gems/LyShine/Code/Source/LyShineSystemComponent.cpp

```diff
@@ -384,16 +384,16 @@ namespace LyShine
         // When module is linked statically, we'll share the application's gEnv pointer.
         gEnv = system.GetGlobalEnvironment();
 #endif
-        m_pLyShine = new CLyShine(gEnv->pSystem);
-        gEnv->pLyShine = m_pLyShine;
+        m_lyShine = AZStd::make_unique<CLyShine>();
+        AZ::Interface<ILyShine>::Register(m_lyShine.get());
```

Comment:
The creation/registration and destruction/unregistration of the CLyShine object can be moved to Activate/Deactivate to remove it from being dependent on CrySystem

Reply:
I have not moved that yet due to gEnv = system.GetGlobalEnvironment(); still being used in LyShine. There are still gEnv references that need to be replaced such as gEnv->pFileIO etc.

Reply:
Only the logic around the `CLyShine` instance would need to be moved at this time.
The rest of `OnCrySystemInitialized` and `OnCrySystemShutdown` can remain as is

Also `gEnv->pFileIO` can be replaced with `AZ::FileIOBase::GetInstance()` since they both point at the same singleton

Reply:
Since I'm not certain if gEnv is accessed during CLyShine creation, I'll leave this as is for now and address the rest of the gEnv refernces in a separate PR.

Reply:
Good with me.